### PR TITLE
Add local user and root user setting up with LibyuiClient

### PR DESCRIPTION
**Pull request by OleksandrOrlov**
Added implementation for user_settings test module with LibyuiClient. The old module is replaced in YaST job group and in opensuse job groups.

Other job groups that use the old module will be addressed in future.

- Related ticket: https://progress.opensuse.org/issues/93624
- Verification run: 
https://openqa.suse.de/tests/6367182, https://openqa.opensuse.org/tests/1818083, https://openqa.opensuse.org/tests/1818084, https://openqa.opensuse.org/tests/1818085, https://openqa.opensuse.org/tests/1818086, https://openqa.opensuse.org/tests/1818087, https://openqa.suse.de/tests/6367707

### CODE REVIEW

File: tests/installation/authentication/default_user_simple_pwd.pm [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+
+# Summary: Add default user with simple password in YaST interactive
+# installation and accept corresponding warning pop-up
+#
+# Maintainer: QA SLE YaST team <qa-sle-yast@suse.de>
+
+use base 'y2_installbase';
+use strict;
+use warnings;
+use scheduler 'get_test_suite_data';
```
Comment (jknphy) [inline tests/installation/authentication/default_user_simple_pwd.pm:18]: not needed anymore.

File: tests/installation/authentication/root_simple_pwd.pm [hunk 2]
```diff
@@ -20,9 +20,9 @@ use Test::Assert ':all';
```
Comment (jknphy) [inline tests/installation/authentication/root_simple_pwd.pm:20]: same here with test data import.

File: tests/installation/yast_firstboot/firstboot_users.pm [hunk 3]
```diff
@@ -19,8 +19,9 @@ use scheduler 'get_test_suite_data';
 sub run {
     my $test_data = get_test_suite_data()->{users};
     my $user_info = {%{$test_data}, (password => $testapi::password)};
-    $testapi::distri->get_local_user()
-      ->create_new_user_with_simple_password($user_info);
+    $testapi::distri->get_local_user()->create_user($user_info);
```
Comment (jknphy) [inline tests/installation/yast_firstboot/firstboot_users.pm:22]: this file looks like should be not needed anymore.

File: tests/installation/authentication/firstboot_user_simple_pwd.pm [hunk 4]
```diff
@@ -0,0 +1,36 @@
...
+
+# Summary: Add firstboot user with simple password in YaST interactive
+# installation and accept corresponding warning pop-up
+#
+# Maintainer: QA SLE YaST team <qa-sle-yast@suse.de>
+
+use base 'y2_installbase';
+use strict;
+use warnings;
+use scheduler 'get_test_suite_data';
```
Comment (jknphy) [inline tests/installation/authentication/firstboot_user_simple_pwd.pm:18]: also here.

File: tests/installation/authentication/enable_autologin.pm [hunk 5]
```diff
@@ -0,0 +1,24 @@
+# SUSE's openQA tests
+#
+# Copyright © 2021 SUSE LLC
+#
+# Copying and distribution of this file, with or without modification,
+# are permitted in any medium without royalty provided the copyright
+# notice and this notice are preserved.  This file is offered as-is,
+# without any warranty.
+
+# Summary: Enable automatic login during user creating
```
Comment (jknphy) [inline tests/installation/authentication/enable_autologin.pm:10]: s/user creating/user creation/

**Review by jknphy — COMMENTED** [review]
LGTM

File: schedule/yast/encryption/activate_encrypted_volume+import_users.yaml [hunk 6]
```diff
@@ -22,7 +22,6 @@ schedule:
   - installation/installer_timezone
   - installation/hostname_inst
   - installation/user_import
-  - installation/user_settings_root
```
Comment (OleksandrOrlov) [inline schedule/yast/encryption/activate_encrypted_volume+import_users.yaml:25]: @jknphy Do we need this change, or we should replace it here with the new test module? (`root_simple_pwd.pm`)

File: schedule/yast/encryption/activate_encrypted_volume+import_users_pvm.yaml [hunk 7]
```diff
@@ -25,7 +25,6 @@ schedule:
   - installation/partitioning_finish
   - installation/installer_timezone
   - installation/user_import
-  - installation/user_settings_root
```
Comment (OleksandrOrlov) [inline schedule/yast/encryption/activate_encrypted_volume+import_users_pvm.yaml:28]: @jknphy Same here, do we need this change, or we should replace it here with the new test module? (`root_simple_pwd.pm`)

**Comment by DimStar77** [comment]
Likely related failures:

* https://openqa.opensuse.org/tests/1821024#

**Comment by OleksandrOrlov** [comment]
@DimStar77 Looking at it. Thanks for pointing out.

**Comment by zhaocongjia** [comment]
Hi @OleksandrOrlov , in "default_user_simple_pwd", you use `full_name => $testapi::realname` explicitly, aka you are using "Bernhard M. Wiedemann" while it used to be "bernhard".  This causes needle mismatch, for example the gdm login screen.  Is this intentional?

**Comment by OleksandrOrlov** [comment]
> Hi @OleksandrOrlov , in "default_user_simple_pwd", you use `full_name => $testapi::realname` explicitly, aka you are using "Bernhard M. Wiedemann" while it used to be "bernhard". This causes needle mismatch, for example the gdm login screen. Is this intentional?

Hi @zhaocongjia , could you please provide me the link to the failed job with this issue?

**Comment by zhaocongjia** [comment]
For example https://openqa.suse.de/tests/7973590#step/first_boot/2, the displayed username is different from previous ones.

**Comment by OleksandrOrlov** [comment]
So, I investigated the issue and here is what I found out:

1. Full name was used until it was changed to `bernhard` in https://github.com/os-autoinst/os-autoinst-distri-opensuse/pull/9808. And it was a workaround, because there was some issue with typing. Now we don't have this issue anymore, as we use [libyui-rest-api](https://github.com/libyui/libyui/tree/master/libyui-rest-api/doc) to communicate with installer. So, the test is returned to its initial state, that allows to check a test case that Full Name is used on login screen. In case same name is used for both Full Name and Username, it is not clear if there is an issue there.
2. As I can see, it was you who created this needle with soft-fail and workaround, without adding a link to Bugzilla ticket (only "shows fullname instead of username" message), so it is not an issue and was never be.

So, as a solution, we should remove a workaround tag from the needle and continue using it as normal needle, as Full Name is used as expected.

**Comment by zhaocongjia** [comment]
Thanks, now I know the background and I'll change the soft-fail needle.

**Pull request merged**